docs/spark/spark-writes.md

```diff
@@ -326,28 +330,61 @@ USING iceberg
 PARTITIONED BY (days(ts), category)
 ```
 
-To write data to the sample table, your data needs to be sorted by `days(ts), category`.
+#### In Spark SQL
 
-If you're inserting data with SQL statement, you can use `ORDER BY` to achieve it, like below:
+To globally sort data based on `ts` and `category`:
 
 ```sql
 INSERT INTO prod.db.sample
 SELECT id, data, category, ts FROM another_table
 ORDER BY ts, category
 ```
 
-If you're inserting data with DataFrame, you can use either `orderBy`/`sort` to trigger global sort, or `sortWithinPartitions`
-to trigger local sort. Local sort for example:
+To locally sort data based on `ts` and `category`:
+
+```sql
+INSERT INTO prod.db.sample
+SELECT id, data, category, ts FROM another_table
+SORT BY ts, category
+```
+
+`SORT BY` clauses can also be used with partition transforms. The [date-and-timestamp-functions](https://spark.apache.org/docs/latest/sql-ref-functions-builtin.html#date-and-timestamp-functions) will be used when partition transforms is time related; truncate related functions such as `substr` will be used when partition transform is `truncate[W]`; [define and register UDFs](https://spark.apache.org/docs/latest/sql-ref-functions-udf-scalar.html) can also take effect.
```

Comment:
This paragraph has a few issues,
`will` -> `should` ? There are multiple "will be used"  when I believe the intent is "can be used" or "should be used"

partition transforms is time related -> are time related // Plural `transforms` needs "are"

related functions such as substr _should_ be used when _the_ partition // "the" before partition and "should" instead of will

The last sentence "define and register udfs can also take effect" doesn't make sense to me in the context of the other sentences. What does this mean in the context of sort by clauses?

Maggie, my resident english expert, also tells me the semi-colons are correct but you could use periods if you liked as well.